servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/Host.java

```diff
@@ -31,6 +31,20 @@
  */
 interface Host<ResolvedAddress, C extends LoadBalancedConnection> extends ListenableAsyncCloseable, ScoreSupplier {
 
+    enum Status {
```

Comment:
i wonder whether this could affect extensibility in the feature by users. Did you consider offering an interface instead?

Reply:
I didn't consider it, but I'm not that excited about it. The constraint on results make it easier for a `Selector` to work for an arbitrary `Host` and vise versa. I also don't see a lot of value in a custom `Host` right now.

If you have specific ideas I'm interested to hear them. If not, we can always change it later: ideally we don't have to make much of this public API other than the configuration bits so it should be a reversible decision for a while.